app/models/credential_factory.rb

```diff
@@ -0,0 +1,49 @@
+module CredentialFactory
+  class << self
+    def values factory_resource
+      provider_annotation = factory_resource.credential_factory_provider
+      raise ArgumentError, %Q(Annotation "credential-factory/provider" not found on #{factory_resource.id.inspect}) unless provider_annotation
+      require "credential_factory/#{provider_annotation.value.underscore}"
+      factory_class = self.const_get(provider_annotation.value.downcase.camelize)
+
+      annotations = factory_resource.annotations.select do |annotation|
```

Comment:
It's not clear to me how that'll work with the database, or if at all. What do you think about using `factory_resource.annotations.where(name: /^credential-factory/)` to make the database do the filtering? Nb, I think it should be pretty harmless to pass `provider` annotation to the provider as well, and -- who knows -- maybe even useful.